benchmarks/benchmarks/pybench/persistence/common.py

```diff
@@ -0,0 +1,90 @@
+import abc
+import datetime
+from pathlib import Path
+from typing import Any, Iterable, Mapping, Protocol, Sequence, TypedDict
+
+from mypy.memprofile import defaultdict
+
+
+class BenchSchemaProto(Protocol):
+    tags: Sequence[str]
+    base: str
+
+class BenchAccessProto(Protocol):
+
+    schemas: Sequence[BenchSchemaProto]
+
+    @abc.abstractmethod
+    def get_name_and_subname(self) -> tuple[str, str]:
+        raise NotImplementedError
+
+    @abc.abstractmethod
+    def bench_result_file(self, bench_id: str) -> Path:
+        raise NotImplementedError
+
+    @abc.abstractmethod
+    def env_spec_str(self) -> str:
+        raise NotImplementedError
+
+    @abc.abstractmethod
+    def get_id(self, schema) -> str:
+         raise NotImplementedError
+
+
+
+class BenchRecord(TypedDict):
+    is_actual: bool
+    benchmark_name: str
+    benchmark_subname: str
+    base: str
+    local_id: str
+    global_id: str
+    tags: str
+    kwargs: str
+    distributions: str
+    data: bytes
+    created_at: datetime.datetime
+
+class BenchWriter(Protocol):
+
+    @abc.abstractmethod
+    def write_bench_data(self, record: BenchRecord) -> None:
+        return
+
+    @abc.abstractmethod
+    def write_release(self, hub_to_director_to_env: Mapping[..., Mapping]) -> None:
+        return
+
+
+class BenchReader(Protocol):
+
+    @abc.abstractmethod
+    def read_schemas_content(self) -> Sequence[str]:
+        raise NotImplementedError
+
+    @abc.abstractmethod
+    def release_to_measures(self, hub_key: str) -> Mapping[..., Sequence[...]]:
+        raise NotImplementedError
+
+    @abc.abstractmethod
+    def get_distributions(self, benchmark_hub: Iterable) -> dict[str, str]:
+        raise NotImplementedError
+
+    @abc.abstractmethod
+    def load_benchmarks(self) -> Sequence:
+        raise NotImplementedError
```

Comment:
`Sequence` means that it returns sequence of any type, but this method returns sequence of dicts